pkg/cloud/rgraph/exec/retrieable_action.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2024 Google LLC
+
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package exec
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+)
+
+// retriableAction is a decorator to the action adding retry mechanism
+type retriableAction struct {
+	Action
+	canRetry func(error) time.Duration
+}
+
+// NewRetriableAction decorates Action with retry function. On error the action
+// will be retried when canRetry(err) returns duration greater than 0.
```

Comment:
We should return (bool, time.Duration). Otherwise if you want to retry immediately (which could be applicable), the user will need to return epsilon time, which doesn't make sense.

Reply:
done